File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int a,b;map<int,int>mp;
        for(int i=0;i<nums.size();i++){
mp[nums[i]]=i;
        }
        for(int i=0;i<nums.size();i++){
        if(mp.find(target-nums[i])!=mp.end() && i!=mp.find(target-nums[i])->second){
            a=i;
            b=mp.find(target-nums[i])->second;
            break;
        }
        }
vector<int> v;
v.push_back(a);v.push_back(b);
        return v;
    }
};

File: Data Structures & Algorithms/target-sum/submission-1.py
class Solution:
    def findTargetSumWays(self, nums: List[int], target: int) -> int:
        n=len(nums)
        S=sum(nums)
        dp=[[0 for _ in range(2*S+1)] for _ in range(n+1)]
        for i in range(2*S+1):
            dp[n][i]=1 if i==target+S else 0
        for i in range(n-1,-1,-1):
            for a in range(2*S+1):
                dp[i][a]=(dp[i+1][a-nums[i]] if a>=nums[i] else 0)+(dp[i+1][a+nums[i]] if a+nums[i]<=2*S else 0)
        return dp[0][S]

        